### ISSUE TITLE
create-astro: Even if installation of dependencies fails, still prints Dependencies installed

### ISSUE THREAD

**Issue by skirianov** [withastro/astro#8990]
### Astro Info

```block
Astro                    v3.4.3
Node                     v18.17.1
System                   macOS (arm64)
Package Manager          unknown
Output                   static
Adapter                  none
Integrations             none
```

### Describe the Bug

While running `create-astro` CLI on slow/bad network you can end up in Timeout error, but current implementation of `spinner` doesn't handle failed cases and will always print success message, even if error was shown. 

This results in this confusing behaviour:
 
<img width="863" alt="Screenshot 2023-11-02 at 22 17 29" src="https://github.com/withastro/astro/assets/74229951/282480b4-52d1-4c10-ad91-76bd6e67a73c">

`spinner` call from `dependencies.ts` file:
<img width="863" alt="Screenshot 2023-11-02 at 22 25 46" src="https://github.com/withastro/astro/assets/74229951/56111445-4c31-4b6c-9584-f34ba7801264">

Maybe it would be nice to handle such edge cases like slow internet, cause I think for my bad internet { cwd, timeout: 90_000, stdio: 'ignore' } timeout of 90s was not enough.

Also, maybe you could use your `isOnline` function every time before doing any network request, just to confirm that the user still has the internet. 

```
function isOnline(): Promise<boolean> {
     return dns.lookup('[http://github.com](https://t.co/dZiD0COfJD)').then(
         () => true,
         () => false
      );
}
```

### What's the expected result?

If due to my internet speed or internet connection interruption I was not able to install dependencies it should result in error. 

Could also print something like 
"Ah, oh, your internet connection is gone."
"Do you want to retry installing dependencies?"

Basically handling this edge case properly. 

### Link to Minimal Reproducible Example

N/A - CLI issue

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by lilnasy** [comment]
Thanks for looking into it!
Do you want to open a PR?

**Comment by skirianov** [comment]
Still digging through the code. It looks like `spinner` itself never meant to be handling errors like this and always returns `end` message which is always success 😅 

```
export async function spinner({ start, end, while: update = () => sleep(100), }, { stdin = process.stdin, stdout = process.stdout } = {}) {
    const act = update();
    const tooslow = Object.create(null);
    const result = await Promise.race([sleep(500).then(() => tooslow), act]);
    if (result === tooslow) {
        const loading = await gradient(chalk.green(start), { stdin, stdout });
        await act;
        loading.stop();
    }
    stdout.write(`${" ".repeat(5)} ${chalk.green("✔")}  ${chalk.green(end)}\n`); // always returns only success message
}
```

So, gotta look into it. 

Wrt to this suggestion: 
> Could also print something like
> "Ah, oh, your internet connection is gone."
> "Do you want to retry installing dependencies?"

Let me know if you want to proceed with it, of course with proper print outs.

**Comment by lilnasy** [comment]
Would it retry installation? I think it would be fine to defer it to the user to do it manually afterwards.

In general, more ambiguous messaging would be better. We don't really know if it is slow/severed internet connection, or if its the server, the package manager, the OS, and the code to determine that info might be a bit much.

**Comment by skirianov** [comment]
Gotcha. Then it will be easier, I'm updating https://github.com/withastro/cli-kit `spinner` code to handle errors and once that is ready, will get to the `create-astro`. So, you can assign it to me :)

**Comment by skirianov** [comment]
@lilnasy fixed with https://github.com/withastro/astro/pull/8993 - this PR depends on https://github.com/withastro/cli-kit/pull/25 ✨

**Comment by lilnasy** [comment]
Closing as fixed by #9048.
